javatests/arcs/core/storage/database/ForeignReferenceManagerTest.kt

```diff
@@ -1,31 +1,84 @@
 package arcs.core.storage.database
 
+import arcs.core.analytics.Analytics
 import arcs.core.data.Schema
 import arcs.core.data.SchemaFields
 import arcs.core.data.SchemaName
+import arcs.core.storage.StorageKey
 import arcs.core.storage.keys.ForeignStorageKey
 import arcs.jvm.storage.database.testutil.FakeDatabase
 import arcs.jvm.storage.database.testutil.FakeDatabaseManager
 import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.junit.runners.JUnit4
+import org.mockito.Mock
+import org.mockito.Mockito.times
+import org.mockito.MockitoAnnotations
 
 @ExperimentalCoroutinesApi
 @RunWith(JUnit4::class)
 class ForeignReferenceManagerTest {
 
-  private val dbManager = FakeDatabaseManager()
-  private val foreignReferenceManager = ForeignReferenceManager(dbManager)
+  @Mock
+  private lateinit var mockAnalytics: Analytics
+  private lateinit var foreignReferenceManager: ForeignReferenceManager
+
+  private var persistentEntitiesCountBefore = 0L
+  private var persistentEntitiesCountAfter = 0L
+  private var nonPersistentEntitiesCountBefore = 0L
+  private var nonPersistentEntitiesCountAfter = 0L
+  private var deletionTriggered = false
```

Comment:
I think it would be nicer if these are inside the dbManager since they are used to drive the fake